internal/xds/translator/jwt.go

```diff
@@ -181,8 +181,10 @@ func buildJWTAuthn(irListener *ir.HTTPListener) (*jwtauthnv3.JwtAuthentication,
 
 // buildXdsUpstreamTLSSocket returns an xDS TransportSocket that uses envoyTrustBundle
 // as the CA to authenticate server certificates.
-func buildXdsUpstreamTLSSocket() (*corev3.TransportSocket, error) {
+// TODO huabing: add support for custom CA and client certificate.
+func buildXdsUpstreamTLSSocket(sni string) (*corev3.TransportSocket, error) {
```

Comment:
Should we use a `BackendRef` also for jwks and make the cluster generator in ext-auth a utility that's also used here? Or do we think that jwks is likely an out-of-cluster endpoint so prefer to not bother the user with FQDN services in this scenario? Either way, if we do that, better to do it in another PR.

Reply:
I believe jwks mostly resides outside of the cluster, so EG shouldn't use a `BackendRef` for it.

Reply:
Sorry, my phrasing was confusing. I agree, no need for backendref here.